exporter/exporterhelper/internal/persistent_storage_batch.go

```diff
@@ -111,6 +111,9 @@ func (bof *batchStruct) getRequestResult(key string) (PersistentRequest, error)
 	if err != nil {
 		return nil, err
 	}
+	if reqIf == nil {
```

Comment:
This is perhaps the most important missing piece in this PR and a cause of the original error

Reply:
Is this a valid case or should getResult return an `errValueNotPresentInBatch` if the value is nil?

Reply:
Some items can be normally not present in the underlying storage, so it is OK for `getResult` to return `nil` and do not consider it error.

At the same time, for certain Get operations, if it is already done, it is anticipated it should yield some result. E.g. if we fetch a specific `PersistentRequest` under a given `key`, we expect some value will be present. Hence in `getRequestResult` we consider it an error. 

Perhaps this requires a comment (I will prepare one) or some more clear handling in code?